lib/js_rails_routes/generator.rb

```diff
@@ -57,15 +62,28 @@ def handle_route(route_name, route_path)
       "export function #{route_name}_path(params) { return process('#{route_path}', params, [#{keys.join(',')}]); }"
     end
 
-    def routes
-      @routes ||= Rails.application.routes.routes
-                       .select(&:name)
-                       .map { |r| [r.name, r.path.spec.to_s.split('(')[0]] }
-                       .sort { |a, b| a[0] <=> b[0] }
+    def routes_with_engines
+      engines = Rails::Engine.subclasses.reject { |klass| klass.to_s.downcase =~ exclude_engines }
+
+      engine_routes = engines.map do |engine|
+        routes = make_routes(engine.routes.routes)
+        [engine.to_s, routes]
+      end
+      engine_routes = engine_routes.reject { |_, routes| routes.empty? }
+      routes = engine_routes.push(['Rails', make_routes(Rails.application.routes.routes)])
+
+      @routes ||= routes
```

Comment:
`@name ||= name` is a idiom in Ruby to cache a result of heavy process. Whenever this method is called, the lines between 66 and 73 are executed. Put these lines in `begin` block to work as expected:

```rb
def routes_with_engines
  @routes_with_engines ||= begin
    # L66-L73 come here
  end
end
```

Reply:
I see. got it.